### PR TITLE
Edit the Source Link spec to point to the new location

### PR DESCRIPTION
**Pull request by chuckries**
This updates the spec to provide an example of generating VSTS source
link. It also adds that VS 2017 15.7 supports authentication.

### CODE REVIEW

**Comment by chuckries** [comment]
@gregg-miskelly @tmat please review.

File: Documentation/diagnostics/source_link.md [hunk 1]
```diff
@@ -133,6 +190,7 @@ In this example:
             }
         }
         ```
+    - Starting in [VS 2017 version 15.7 Preview 3](https://blogs.msdn.microsoft.com/visualstudio/2018/04/09/visual-studio-2017-version-15-7-preview-3/), the debugger can automatically authenticate source link requests for VSTS and GitHub repositories.
 - [C# Extension for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp)
     - Source Link is a planned feature but is not yet supported.
```
Comment (gregg-miskelly) [inline Documentation/diagnostics/source_link.md]: Can you fix this part also while you are updating? It should say: "Supports Source Link as of C# extension version 1.15"

File: Documentation/diagnostics/source_link.md [hunk 2]
```diff
@@ -7,41 +7,98 @@ Most debugging is done against locally built source code on a developer's machin
 ## Source Link File Specification
 The `/sourcelink:<file>` compiler flag will embed a JSON configuration file in the PDB. This configuration file would be generated as part of the build process. The JSON configuration file contains a simple mapping of local file path to URL where the source file can be retrieved via http or https. A debugger would retrieve the original file path of the current location from the PDB, look this path up in the Source Link map, and use the resulting URL to download the source file.
 
-### MSBuild Example
-There is an open source tool available for generating Source Link configuration at build time. See https://github.com/ctaggart/SourceLink for the most up to date information.
+## MSBuild Example
+There is an open source tool available for generating Source Link configuration at build time. See https://github.com/ctaggart/SourceLink#quick-start for the most up to date information.
```
Comment (gregg-miskelly) [inline Documentation/diagnostics/source_link.md]: After the first sentence, should we add "This supports GitHub and BitBucket out of the box"?

**Comment by chuckries** [comment]
new PR against the designs repo: https://github.com/dotnet/designs/pull/37

I have changed this PR to just update the doc to be a link to the new location.

**Comment by chuckries** [comment]
@gregg-miskelly I have update https://aka.ms/sourcelinkspec to point to the new location as well.

**Pull request merged**